### PR TITLE
RFD animations for building stuff

### PR DESCRIPTION
**Pull request by alsoandanswer**
title
https://camo.githubusercontent.com/ddcdeb42d126a3e0ced90d3296dd9a6154a017af/68747470733a2f2f692e696d6775722e636f6d2f726f78356c36652e676966

IT WORKS

However, walls don't play the end animation (just obscured) because the ways walls spawn are an assfuck, and will probably work properly once the code that spawns walls doesn't suck.

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/effects/misc.dm [hunk 1]
```diff
@@ -37,3 +37,44 @@
...
+/obj/effect/constructing_effect/proc/end_animation()
+	if(status == 3)
+		end()
+	else
+		icon_state = "rfd_end"
+		addtimer(CALLBACK(src, .proc/end), 15)
+
+/obj/effect/constructing_effect/proc/end()
+	qdel(src)
+s
```
Comment (NonQueueingMatt) [inline code/game/objects/effects/misc.dm:80]: s

File: code/game/objects/items/weapons/RFD.dm [hunk 2]
```diff
@@ -150,43 +154,38 @@ RFD Construction-Class
...
-	if(working == 1)
-		return 0
+	if(working)
+		return FALSE
 
 	if(mode == 3 && istype(T,/obj/machinery/door/airlock))
-		build_cost =  10
-		build_delay = 50
+		build_cost =  3
+		build_delay = 20
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/RFD.dm:182]: undocumented change

Comment (alsoandanswer) [reply code/game/objects/items/weapons/RFD.dm:182]: Addressed, added to changelog

File: code/game/objects/items/weapons/RFD.dm [hunk 3]
```diff
@@ -203,39 +202,50 @@ RFD Construction-Class
...
 	else
 		qdel(T)
 
+	rfd_effect.end_animation()
 	playsound(get_turf(src), 'sound/effects/magnetclamp.ogg', 50, 1)
-	return 1
+	build_cost = null
+	build_delay = null
+	build_type = null
+	build_turf = null // So it resets and any fuckery is avoided
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/RFD.dm:246]: why's this on the obj itself rather than in the proc?

Comment (alsoandanswer) [reply code/game/objects/items/weapons/RFD.dm:245]: Actually this doesn't relate to RFD animations.

Apparently RFDs can't deconstruct airlocks if you swap between modes.

Unless you do this.

**Pull request merged**